### PR TITLE
cluster feature extension guide

### PR DESCRIPTION
**Pull request by jim-docker**
Note: also modified one source file, comments only

All code added has been tested
Still need to complete everything below WIP heading

addresses #1240

### CODE REVIEW

File: docs/extensions/guides/README.md [hunk 1]
```diff
@@ -24,5 +24,9 @@ Each guide or sample will include:
 
 | Sample | APIs |
 | ----- | ----- |
+[custom-resource-page](https://github.com/lensapp/lens-extension-samples/tree/master/custom-resource-page) | LensRendererExtension <br> K8sApi.KubeApi <br> K8sApi.KubeObjectStore <br> Component.KubeObjectListLayout <br> Component.KubeObjectDetailsProps <br> Component.IconProps |
```
Comment (Nokel81) [inline docs/extensions/guides/README.md]: Should "helloworld" be first?

File: docs/extensions/guides/renderer-extension.md [hunk 2]
```diff
@@ -317,6 +306,87 @@ The `title` and `components.Description` fields appear on the cluster settings p
   abstract updateStatus(cluster: Cluster): Promise<ClusterFeatureStatus>;
 ```
 
+The `install()` method is typically called by Lens when a user has indicated that this feature is to be installed (i.e. clicked **Install** for the feature on the cluster settings page). The implementation of this method should install kubernetes resources using the `applyResources()` method, or by directly accessing the kubernetes api ([`K8sApi`](tbd)).
```
Comment (Nokel81) [inline docs/extensions/guides/renderer-extension.md:309]: I would leave out the word "typically"

**Review by Nokel81 — APPROVED** [review]
The improvements look good. Do we still want the "WIP" heading though?

**Comment by jim-docker** [comment]
> The improvements look good. Do we still want the "WIP" heading though?

Thanks. Everything below the WIP heading still needs work (what's there came straight from common-capabilities :) )

File: docs/extensions/guides/renderer-extension.md (hunk 2 above)
Comment (jim-docker) [reply docs/extensions/guides/renderer-extension.md:309]: I want to convey that it is not necessarily the only way `install()` gets called (Lens could now or in the future invoke it in another context. Could be invoked by other extension code...) Maybe there's a better word or phrase (can't be very long though)

File: docs/extensions/guides/README.md (hunk 1 above)
Comment (jim-docker) [reply docs/extensions/guides/README.md]: Probably? It's just alphabetical for now. I figure wait till the list is complete

Comment (jim-docker) [reply docs/extensions/guides/README.md]: moved it while resolving merge conflict

**Pull request merged**